src/main/java/io/confluent/clientmetrics/MetricsPublisher.java

```diff
@@ -0,0 +1,102 @@
+package io.confluent.clientmetrics;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.clientmetrics.avro.AVROMetricRecords;
+import io.confluent.clientmetrics.avro.AVROMetricValue;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Metrics publisher can be scheduled to run periodically, like:
+ * <p>
+ * final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
+ * executorService.scheduleAtFixedRate(() -> sendMetrics(producer, streamsClient), 0, 1, TimeUnit.SECONDS);
+ */
+public class MetricsPublisher<V> {
+
+    private static final String DEFAULT_METRICS_TOPIC = "_confluent_client_metrics";
+
+    private final String clientMetricsTopic;
+    private final MetricsSender<V> metricsSender;
+
+    public MetricsPublisher(MetricsSender<V> metricsSender) {
```

Comment:
implement a default sender

Reply:
Why? I do not think I like default hidden assignments. Personally, I would prefer to keep it explicit.

Reply:
Just so it makes it easier to use for beginners. Make an opinion, don't be shy.. the user can switch constructors if they want to use their own explicit one.
Sorry, i wasn't clear, i was suggesting to add another constructor which takes no params, which takes no params..